src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -150,6 +183,158 @@ def set_server_facts
+        # its configuration
+        indirection_instance = Puppet::Indirector::Indirection.instance(indirection)
+
+        # An actual class ref of what will be indirected, eg `Puppet::Node::Facts`
+        indirected_class_reference = indirection_instance.model
+
+        # Symbol or String (eg :store_configs) or `nil` (no cache)
+        # non-nil can be used to look up class ref eg
+        # `Terminus.terminus_class(:catalog, :store_configs)`
+        cache_terminus_name = indirection_instance.cache_class
+
+        # The symbol, eg :compiler that can be given to
+        # `Terminus.terminus_class` same as cache_class
+        # May be `nil`, if so terminus_setting should be consulted
+        primary_terminus_name = indirection_instance.terminus_class
+
+        # Where to find any configuration for what default terminus to use
+        # Will be a symbol that can be passed into `Puppet.setting[<here>]`
+        terminus_setting = indirection_instance.terminus_setting
+
+        return indirected_class_reference, cache_terminus_name,
+          primary_terminus_name, terminus_setting
+      end
+
+      # Returns a Hash with symbol keys naming each indirection (eg :facts)
+      # Each indirection key refers to a Hash with the configured termini
+      # (name and class) for its primary and cache usages, if a termini is
+      # :store_configs it will find the associated termini for
+      # :storeconfigs_backend and place that in the "actual" keys for the
+      # termini. If not store_configs, then the actual keys will be the
+      # values in origal "configured" keys. Also inlcudes all possible termini
+      # for the given indirection, and additional information needed to derive
+      # the above. eg:
+      #   {
+      #     :catalog =>
+      #       {:indirected_class       => Puppet::Resource::Catalog,
+      #        :cache_terminus_name    => :store_configs,
+      #        :actual_cache_name      => :puppetdb,
+      #        :primary_terminus_name  => :compiler,
+      #        :actual_terminus_name   => :compiler,
+      #        :cache_terminus_class   => Puppet::Resource::Catalog::StoreConfigs,
+      #        :actual_cache_class     => Puppet::Resource::Catalog::Puppetdb,
+      #        :primary_terminus_class => Puppet::Resource::Catalog::Compiler,
+      #        :actual_terminus_class  => Puppet::Resource::Catalog::Compiler,
+      #        :terminus_setting       => :catalog_terminus,
+      #        :termini=>
+      #         {:compiler      => Puppet::Resource::Catalog::Compiler,
+      #          :json          => Puppet::Resource::Catalog::Json,
+      #          :msgpack       => Puppet::Resource::Catalog::Msgpack,
+      #          :rest          => Puppet::Resource::Catalog::Rest,
+      #          :store_configs => Puppet::Resource::Catalog::StoreConfigs,
+      #          :yaml          => Puppet::Resource::Catalog::Yaml,
+      #          :puppetdb      => Puppet::Resource::Catalog::Puppetdb}},
```

Comment:
Do we have a use for this list yet? I don't mind it being here, just curious.

Reply:
No, I started just collecting the info that the indirection knew about. Was useful when building this. But I don't think it'd be needed unless we let people choose specific termini to use. I can remove it.